strings/longest_substr_without_repeating_chars.cpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file
+ * @brief Greedy algorithm to find length of longest substring without repeating
+ * characters. The following
+ * [resource](https://www.geeksforgeeks.org/length-of-the-longest-substring-without-repeating-characters/)
+ * was referred to
+ *
+ *
+ * @details
+ * We are given a string. We need to compute the length of the longest
+ * substring, such that no character in the substring occurs more than once.
+ *
+ * For a string of length n, the time complexity of this algorithm is O(n)
+ * (amortized)
+ *
+ * @author [Nishant Mahesh](https://github.com/nishant-mahesh)
+ */
+
+#include <algorithm>  // To use the inbuilt 'max' function
+#include <cassert>
+#include <iostream>
+#include <map>
+#include <string>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```